docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -474,9 +474,18 @@ Users can instruct the container to not remove any of their specific beans (even
  
 Furthermore, extensions can eliminate possible false positives by producing `UnremovableBeanBuildItem`.
 
+The container will also remove bean producing methods (methods annotated with `@Produces`) from producer beans if the method produces an unused bean.
+You can also annotate bean producing methods with `io.quarkus.arc.Unremovable`.
```

Comment:
I'd append this to the sentence above starting `Users can instruct...` in a following way:

```
Users can instruct the container to not remove any of their specific beans (even if they satisfy all the rules specified above) by annotating them with `io.quarkus.arc.Unremovable`.
This applies to all forms of bean declarations - bean class, producer method, producer field.
```

Would that be clear enough?

Reply:
I rephrase it a little to be crystal clear by splitting it to two sentences:

- One explaining that the optimization (removing beans) apply to all sort of bean definition
- One explaining that the annotation can be put on all sort of bean definition

I think that it worth being explicit in case the developer don't know the spec very well (like me more used to Spring DI than CDI)